metrics/collector_test.go

```diff
@@ -121,6 +121,52 @@ func testMachineStatus(t *testing.T) {
 			if !found {
 				t.Error("metrics sabakan_machine_status was not found")
 			}
+
+			// If machines are deleted, corresponding metrics is also deleted
+			err = model.Machine.SetState(context.Background(), "001", sabakan.StateRetiring)
+			if err != nil {
+				t.Fatal(err)
+			}
+			err = model.Machine.SetState(context.Background(), "001", sabakan.StateRetired)
+			if err != nil {
+				t.Fatal(err)
+			}
+			err = model.Machine.Delete(context.Background(), "001")
+			if err != nil {
+				t.Fatal(err)
+			}
+			err = model.Machine.SetState(context.Background(), "002", sabakan.StateRetiring)
+			if err != nil {
+				t.Fatal(err)
+			}
+			err = model.Machine.SetState(context.Background(), "002", sabakan.StateRetired)
+			if err != nil {
+				t.Fatal(err)
+			}
+			err = model.Machine.Delete(context.Background(), "002")
```

Comment:
Why do you delete both "001" and "002" ?
I would make sure that the metrics of the deleted machine disappear, and that the metrics of the undeleted machine **do not** disappear.